### PR TITLE
proxy options feature for callouts

### PR DESCRIPTION
**Pull request by marcellcsiszar**
I really appreciate your work. However, I need to access resources through enterprise proxy, so I've added proxy related options. Finally I can work on some legacy app decommissioning projects. I hope you will like that

### CODE REVIEW

File: lib/install.js [hunk 1]
```diff
@@ -81,7 +88,7 @@ async function install(targetPath, version) {
   }
 }
 
-module.exports = async function(version) {
+module.exports = async function(version,proxy) {
```
Comment (jchip) [inline lib/install.js]: you had a 3rd param `verifyssl` that you were passing to this?

File: lib/install.js [hunk 2]
```diff
@@ -103,7 +110,7 @@ module.exports = async function(version) {
 
   mkdirp.sync(path.join(common.getNvmCacheDir(), version));
 
-  const status = await downloadNode(version);
+  const status = await downloadNode(version,proxy);
```
Comment (jchip) [inline lib/install.js]: need to pass the `verifyssl` param

Comment (marcellcsiszar) [reply lib/install.js]: forget to include it in this branch, great catch thanks!

File: lib/install.js (hunk 1 above)
Comment (marcellcsiszar) [reply lib/install.js]: forget to include it in this branch, great catch thanks!

Comment (marcellcsiszar) [reply lib/install.js]: added

File: lib/install.js (hunk 2 above)
Comment (marcellcsiszar) [reply lib/install.js]: added

**Pull request merged**